acme/api/handler.go

```diff
@@ -145,12 +153,27 @@ func (d *Directory) ToLog() (interface{}, error) {
 // for client configuration.
 func (h *Handler) GetDirectory(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
+	prov, err := provisionerFromContext(ctx)
+	if err != nil {
+		api.WriteError(w, err)
+		return
+	}
+
+	acmeProv, ok := prov.(*provisioner.ACME) // TODO: rewrite into providing configuration via function on acme.Provisioner
```

Comment:
We may want to just add this section to `provisionerFromContext`, right? Is there anywhere where we load a provisioner using this method where we don't want to make sure it's an ACME provisioner? We only use this method in the ACME code. But if we want to be extra safe, we could just make it a new method like `acmeProvisionerFromContext`.

Reply:
The `provisionerFromContext` is indeed only used in the ACME part (there's another, similar function for SCEP, though), but I think the idea was to provide just the interface for reasons related to overall architecture. the `RequireEAB` is the first property that is used directly and didn't really fit in with the way other properties for the (ACME) provisioner were handled, so I chose to implement `acmeProvisionerFromContext`. There're probably other ways, like having a function on the `acme.Provisioner` interface; perhaps I'll think of something nicer.